src/common/schedulers/slurm_commands.py

```diff
@@ -467,53 +667,34 @@ def _parse_nodes_info(slurm_node_info):
     for node in node_info:
         lines = node.strip().splitlines()
         if lines:
-            slurm_nodes.append(SlurmNode(*lines))
+            if is_static_node(lines[0]):
+                node = StaticNode(*lines)
+                slurm_nodes.append(node)
+            else:
+                node = DynamicNode(*lines)
+                slurm_nodes.append(node)
     return slurm_nodes
 
 
-def get_nodes_type(nodes):
-    """Get node type given nodename. Node type format: NodeType(instancetype, queue_name)."""
-    nodes_types = set()
-    for node in nodes:
-        queue_name, _, instance_name = parse_nodename(node.name)
-        nodes_types.add(NodeType(instance_name, queue_name))
-    return nodes_types
-
-
-def get_inactive_partition_names(partitions):
+def get_inactive_partition_names(partition_names_map):
     """Get the name of all inactive partitions from partitions and name mapping."""
     inactive_partitions_name = []
-    for partition_name, partition in partitions.items():
+    for partition_name, partition in partition_names_map.items():
         if partition.state == "INACTIVE":
             inactive_partitions_name.append(partition_name)
     return inactive_partitions_name
 
 
-def retrieve_partitions_from_node_types(node_types):
-    """Retrieve partitions for node types."""
-    return {node_type.partition for node_type in node_types}
-
-
-def get_partition_node_names(nodenames):
-    """
-    Convert partition nodenames read from partition to a list of nodes.
-
-    Example input nodenames: "queue1-st-c5xlarge-[1,3,4-5],queue1-st-c5large-20"
-    Example output [queue1-st-c5xlarge-1, queue1-st-c5xlarge-3, queue1-st-c5xlarge-4, queue1-st-c5xlarge-5,
-    queue1-st-c5large-20]
-    """
-    if type(nodenames) is str:
-        matches = re.findall(r"((([a-z0-9\-]+)-(st|dy)-([a-z0-9]+)-)(\[[\d+,-]+\]|\d+))", nodenames)
-        # [('queue1-st-c5xlarge-[1,3,4-5]', 'queue1-st-c5xlarge-', 'queue1', 'st', 'c5xlarge', '[1,3,4-5]'),
-        # ('queue1-st-c5large-20', 'queue1-st-c5large-', 'queue1', 'st', 'c5large', '20')]
-    partition_node_names = []
-    for match in matches:
-        nodename, prefix, _, _, _, nodes = match
-        if "[" not in nodes:
-            # Single nodename
-            partition_node_names.append(nodename)
-        else:
-            # Multiple nodenames
-            node_list = convert_range_to_list(nodes.strip("[]"))
-            partition_node_names += [prefix + str(n) for n in node_list]
-    return partition_node_names
+def time_is_up(initial_time, current_time, grace_time):
```

Comment:
As we discussed this morning, we probably don't want to put this in `slurm_commands`